jina/types/arrays/neural_ops.py

```diff
@@ -82,3 +83,26 @@ def match(
                 d = Document(darray[int(_id)], copy=True)
                 d.scores[m_name] = _dist
                 _q.matches.append(d)
+
+    def visualize(
+        self, colors: Union[None, np.ndarray] = None, file_path: Union[None, str] = None
+    ):
+        """Visualize embeddings in a 2D projection with the PCA algorithm.
+
+        :param colors: Optional np.ndarray of lengh equal to len(self) containing integers for colors.
+        :param file_path: Optional path to store the visualization.
+        """
+
+        import matplotlib.pyplot as plt
+
+        pca = PCA(n_components=2)
+        x_mat = np.stack(self.get_attributes('embedding'))
+        x_mat_2d = pca.fit_transform(x_mat)
+
+        plt.scatter(x_mat_2d[:, 0], x_mat_2d[:, 1], c=colors)
```

Comment:
i think it would be useful to make colors a mapping of `doc.tags['some_tag']`, e.g.

basically implement logic to enable the following mapping
```
if tags['dummy_label'] == 'dummy_value': color = 1
if tags['dummy_label'] == 'dummy_value2': color = 2
```

any opinions?

Reply:
use dot with alpha value gives better visualization

Reply:
I've updated the code with what you propose, coloring by tag makes sense